src/test/collectionHandlingFunctions.php

```diff
@@ -0,0 +1,314 @@
+<?php
+require "../collectionHandlingFunctions.php";
+use PHPUnit\Framework\TestCase;
+
+class collectionHandlingFunctions extends TestCase 
+{
+
+    public function testFailureDescribeCollectionItem_EmptyItem() 
+    {
+        $yarnExample = [];
+        $this->expectException(Exception::class); 
+        describeCollectionItem($yarnExample);
+    }
+    
+    public function testFailureDescribeCollectionItem_undefinedName() {
+        $yarnExample = ["name" => NULL, 
+                        "shade" => "Rainbow", 
+                        "dominantColour" => "purple",
+                        "composition" => "Cashmere",
+                        "yarnType" => "Super Chunky", 
+                        "lengthInMeters" => 100,
+                        "skeinNumber" => 2, 
+                        "project"=> "Cool rainbow hat",
+                        "image" => "blue_scarf.jpg"      
+                    ];
+        $this->expectException(Exception::class);
+        describeCollectionItem($yarnExample);
+    }
+
+    public function testSuccessDescribeCollectionItem_EverythingDefined()
+    {
+        $yarnExample = ["name" => "Unicorn brand", 
+                        "shade" => "Rainbow", 
+                        "dominantColour" => "purple",
+                        "composition" => "Cashmere",
+                        "yarnType" => "Super Chunky", 
+                        "lengthInMeters" => 100,
+                        "skeinNumber" => 2, 
+                        "project"=> "Cool rainbow hat",
+                        "image" => "blue_scarf.jpg"      
+                    ];
+        $expectedOutput = '<div class="collection_item">';
+        $expectedOutput .= '<h3 tabindex="0">Unicorn brand Rainbow</h3>';
+        $expectedOutput .= '<img tabindex="0" src="images/blue_scarf.jpg" alt="A picture of Unicorn brand in shade Rainbow">';              
+        $expectedOutput .= '<h4 tabindex="0">Description</h4>';
+        $expectedOutput .= '<ul>';
+        $expectedOutput .= '<li>Dominant colour: purple</li>';
+        $expectedOutput .= '<li>Yarn type: Super Chunky</li>';
+        $expectedOutput .= '<li>Composition: Cashmere</li>';
+        $expectedOutput .= '<li>Length of a skein: 100 m</li>';            
+        $expectedOutput .= '<li>Quantity: 2 skeins in stash</li>';
+        $expectedOutput .= '<li>Allocated to project: Cool rainbow hat</li>';
+        $expectedOutput .= '</ul>';
+        $expectedOutput .= '</div>'; 
+        
+        $actualOutput = describeCollectionItem($yarnExample);
+
+        $this->assertEquals($expectedOutput, $actualOutput);
+    }
+
+ //The following functions check that the output is correct when one of the fields is missing 
+ // that item in the database    
+    public function testSuccessDescribeCollectionItem_undefinedShade()
```

Comment:
the comments should be in line with the function indentation wise

Reply:
Done